domain/application/service/application.go

```diff
@@ -145,27 +147,46 @@ var (
 	validApplication = regexp.MustCompile("^" + applicationSnippet + "$")
 )
 
+// SecretService defines methods on a secret service
+// used by an application service.
+type SecretService interface {
+	InternalDeleteSecret(ctx domain.AtomicContext, uri *coresecrets.URI, params secretservice.DeleteSecretParams) (func(), error)
+}
+
+// NotImplementedSecretService defines a secret service which does nothing.
+type NotImplementedSecretService struct{}
+
+func (NotImplementedSecretService) InternalDeleteSecret(domain.AtomicContext, *coresecrets.URI, secretservice.DeleteSecretParams) (func(), error) {
+	return func() {}, nil
```

Comment:
Do we want to fail it or just a no ops.

Reply:
just no op because not all users of the service need to deal with secrets, eg some users just want to set a charm url or whatever